api/src/main/java/org/openmrs/api/db/hibernate/HibernateObsDAO.java

```diff
@@ -255,5 +255,14 @@ public Obs getObsByUuid(String uuid) {
 		return (Obs) sessionFactory.getCurrentSession().createQuery("from Obs o where o.uuid = :uuid").setString("uuid",
 		    uuid).uniqueResult();
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.ObsDAO#getRevisionObs(org.openmrs.Obs)
+	 */
+	public Obs getRevisionObs(Obs initialObs) {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Obs.class, "obs");
+		criteria.add(Restrictions.eq("previousVersion", initialObs));
+		return (Obs) criteria.list().get(0);
```

Comment:
Shouldn't this be a unique result? An order should have exactly one previous order to it